netgrasp/utils/mac.py

```diff
@@ -0,0 +1,255 @@
+
+    mac = normalize_mac(mac)
+    debugger.debug('looking up vendor for %s' % mac)
+    http = httplib.HTTPConnection('api.macvendors.com', port=80)
+    url = '/%s' % mac
+    http.request('GET', url)
+    response = http.getresponse()
+    if response.status == httplib.OK:
+        vendor = response.read()
+        debugger.debugger.debug('identified %s as %s' % (mac, vendor))
+    else:
+        vendor = None
+        debugger.debugger.info('failed to identify %s' % mac)
+
+    return vendor
+
+
+def print_lookup_config(dbfile, lockfile, debugger):
+    ok, config = get_lookup_config(dbfile, lockfile, debugger)
+    if not ok:
+        print(config)
+        return
+    print('mac lookup configuration')
+    print('source: %s' % config['source'])
+    print('mac database was last updated: %s' % get_dbupdated_text(config['updated']))
+
+
+def change_source(dbfile, lockfile, debugger, source):
+    updated_msg_never = 'run "netgrasp mac -u" to update mac database'
+    updated_msg_remind = 'if too far out of date, run "netgrasp mac -u" to update mac database'
+
+    ok, config = get_lookup_config(dbfile, lockfile, debugger)
+    if not ok:
+        print(config)
+        return
+    # Write new config with source. Keep existing updated timestamp.
+    updated = config['updated']
+    ok, wrcfg = write_lookup_config(dbfile, lockfile, debugger, source=source, updated=updated)
+    if not ok:
+        print(wrcfg)
+        return
+    print('source successfully set to "%s"' % source)
+    if source == 'local':
+        print('mac database was last updated: %s' % get_dbupdated_text(updated))
+        if updated == DEFAULT_LOOKUP_UPDATED:
+            print(updated_msg_never)
+        else:
+            print(updated_msg_remind)
+
+
+def update_db(dbfile, lockfile, debugger):
+    print('starting mac database update')
+    try:
+        if TEST_WSDATA:
+            with open(os.path.expanduser(TEST_WSDATA_FILENAME)) as f:
+                wsdata = f.read()
+            wsdata = wsdata.decode('utf-8', errors='replace')
+        else:
+            wsdata = download_wireshark_data(debugger)
+        dbdata = parse_wireshark_data(wsdata)
+        numrecs = len(dbdata)
+        if numrecs:
+            debugger.info('parsed %d records for import' % numrecs)
+        else:
+            raise Exception('parse error: %d records parsed for import' % numrecs)
+        import_data(dbdata, dbfile, lockfile, debugger)
+    except Exception as e:
+        debugger.dump_exception('mac database update failed')
+        print('mac database update failed: %s' % e)
+    else:
+        print(WIRESHARK_ATTRIBUTION)
+        print('mac database update successful')
+
+
+def get_lookup_config(dbfile, lockfile, debugger):
+    from netgrasp.database import database
+
+    try:
+        db = database.Database(dbfile, lockfile, debugger)
+        source = db.get_state(DB_LOOKUP_CONFIG_SOURCE_KEY)
+        updated = db.get_state(DB_LOOKUP_CONFIG_UPDATED_KEY, is_date=True)
```

Comment:
I find this define abstraction more confusing: I'd rather easily/quickly see what's being queried rather than have to look it up if debugging/working in this section of the code.